pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IngestionDelayTracker.java

```diff
@@ -139,17 +166,44 @@ private static class IngestionInfo {
 
   private Clock _clock;
 
+  // Configuration parameters
+  private final boolean _enableOffsetLagMetric;
+  private final long _offsetLagUpdateIntervalMs;
+
+  private final StreamConsumerFactory _streamConsumerFactory;
+
   @VisibleForTesting
   public IngestionDelayTracker(ServerMetrics serverMetrics, String tableNameWithType,
       RealtimeTableDataManager realtimeTableDataManager, int scheduledExecutorThreadTickIntervalMs,
-      Supplier<Boolean> isServerReadyToServeQueries)
+      Supplier<Boolean> isServerReadyToServeQueries, TableConfig tableConfig)
       throws RuntimeException {
     _serverMetrics = serverMetrics;
     _tableNameWithType = tableNameWithType;
     _metricName = tableNameWithType;
     _realTimeTableDataManager = realtimeTableDataManager;
     _clock = Clock.systemUTC();
     _isServerReadyToServeQueries = isServerReadyToServeQueries;
+
+    StreamConfig streamConfig =
```

Comment:
Ideally we should directly pass in `StreamConfig` to avoid parsing it multiple times